simapp/v2/upgrades.go

```diff
@@ -10,20 +10,24 @@ import (
 	epochstypes "cosmossdk.io/x/epochs/types"
 	protocolpooltypes "cosmossdk.io/x/protocolpool/types"
 	upgradetypes "cosmossdk.io/x/upgrade/types"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
 )
 
 // UpgradeName defines the on-chain upgrade name for the sample SimApp upgrade
-// from v0.50.x to v2
+// from v0.50.x to v052
```

Comment:
Nit, this should stay v2

Reply:
oh you mean we can upgrade v0.50.x directly to v2 without upgrade to v.052.x first?

Reply:
Hey yes so that's the difference between simapp and simapp/v2

Basically here simapp/v2 should contains the changes from 0.52 + creating an stf key.

Simapp/v2 is basiscally using v2 for the core layer. But apart from the core layer, SDK 0.52 is just used in an v2 app

Reply:
Got it thanks, there's upgrade overlap work in v0.52.x, so both v0.52.x and main branch need use v050-to-v2?

Reply:
Theoretically main branch only needs v0.52-v2 (which should only be the stf key addition) but 0.52 branch needs 0.50-0.52/v2 yeah